[ZK filter] Add missing operation supports for multi opcode

Winbobob opened the pull request:
Commit Message: [ZK filter] Add missing operation supports for multi opcode
Additional Description: The ZK filter did not support some operations for the multi opcode based on the [ZK source code](https://github.com/apache/zookeeper/blob/58eed9f5280be1c6a9ccacc47dd6afa65e916ae8/zookeeper-server/src/main/java/org/apache/zookeeper/MultiOperationRecord.java#L97-L105). This could lead to decode errors. This PR tries to fill this gap by adding the opcode support of `Create2`, `CreateContainer`, `CreateTtl`, `GetChildren`, and `GetData` for the `Multi` operation.
Risk Level: Low
Testing: Unit tests
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

Winbobob commented:
cc @JuniorHsu, please review if you have a chance, thank you!

JuniorHsu commented on source/extensions/filters/network/zookeeper_proxy/decoder.cc:
```diff
@@ -642,22 +642,34 @@ absl::Status DecoderImpl::parseMultiRequest(Buffer::Instance& data, uint64_t& of
     }
 
-    switch (static_cast<OpCodes>(opcode.value())) {
+    const auto op = static_cast<OpCodes>(opcode.value());
```
any motivation for one more variable? I'll prefer one less stack memory if we don't have good reason

JuniorHsu replied:
oh i see you need it for the macro so ignore this

JuniorHsu commented on source/extensions/filters/network/zookeeper_proxy/decoder.cc:
```diff
@@ -642,22 +642,34 @@ absl::Status DecoderImpl::parseMultiRequest(Buffer::Instance& data, uint64_t& of
     case OpCodes::Create:
-      status = parseCreateRequest(data, offset, len, OpCodes::Create);
-      EMIT_DECODER_ERR_AND_RETURN_IF_STATUS_NOT_OK(status, OpCodes::Create);
+    case OpCodes::Create2:
```
`FALLTHRU;` between case: so we know it's intentional

JuniorHsu reviewed (changes requested):
only one nits. I'm going to urge you to express what OP code is supported by this PR instead of one single link in the description

Winbobob commented:
> only one nits. I'm going to urge you to express what OP code is supported by this PR instead of one single link in the description

Just updated the desc with more details and addressed your comments, PTAL!

Winbobob commented:
/retest

JuniorHsu reviewed (approved):
thanks
